app/src/test/java/org/mozilla/fenix/components/metrics/AppStartupTelemetryTest.kt

```diff
@@ -0,0 +1,163 @@
+package org.mozilla.fenix.components.metrics
+
+import android.content.Intent
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.verify
+import mozilla.components.support.utils.toSafeIntent
+import org.junit.Before
+import org.junit.Test
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.LINK
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.CUSTOM_TAB
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.APP_ICON
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Source.UNKNOWN
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.COLD
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.WARM
+import org.mozilla.fenix.components.metrics.Event.AppStartupSourceAndType.Type.HOT
+
+class AppStartupTelemetryTest {
+
+    @RelaxedMockK
+    private lateinit var metricController: MetricController
+    @RelaxedMockK
+    private lateinit var intent: Intent
+
+    private lateinit var appStartupTelemetry: AppStartupTelemetry
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this)
+        appStartupTelemetry = AppStartupTelemetry(metricController)
+    }
+
+    @Test
+    fun `WHEN application is launch for the first time through application icon THEN startup metric records {APP_ICON, COLD, FALSE}`() {
+        every { intent.action } returns Intent.ACTION_MAIN
+        every { intent.categories.contains(Intent.CATEGORY_LAUNCHER) } returns true
```

Comment:
it'd be less fragile to have categories return a set with `CATEGORY_LAUNCHER` rather than mock `categories`

Reply:
I tried something like that but was getting weird mock error for some reason... ill give it another shot